### ISSUE TITLE
coq-insert-intros: missing whitespace after period for multiple autogenerated intros

### ISSUE THREAD

**Issue by gnuvan** [ProofGeneral/PG#1]
This issue was originally detected in company-coq for its `intros!' feature:
https://github.com/cpitclaudel/company-coq/issues/23
It turns out that it is already present in ProofGeneral (coq-insert-intros).

When coq-insert-intros generates several calls to intros, they are separated with a period without whitespace.

Consider the following Coq script (23 True hypotheses imply False):

Goal True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> False.

Now, when doing coq-insert-intros, here is the generated sequence of tactic calls:

intros H H0 H1 H2 H3 H4 H5 H6 H7 H8 H9 H10 H11 H12 H13 H14 H15 H16 H17 H18 H19 H20.intros H21.

There should be a whitespace after "H20." so that intros can be considered as a tactic call rather than a name qualified by the "H20" namespace.

In other words, there should be a whitespace after each "intros ... ." tactic line generated by coq-insert-intros for each line output by Show Intros.

(This seemingly small bug can be annoying for autogenerated lemmas with many hypotheses.)

**Comment by Matafou** [comment]
I will have a look at it.
P.

2015-09-28 16:55 GMT+02:00 gnuvan <notifications@github.com>:
